mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/InsertTest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+
+package org.hibernate.ogm.datastore.mongodb.test;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+
+import org.hibernate.ogm.datastore.mongodb.utils.MongoDbServerRunner;
+import org.hibernate.ogm.utils.OgmTestCase;
+import org.hibernate.ogm.utils.TestForIssue;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author The Viet Nguyen
+ */
+@TestForIssue(jiraKey = "OGM-1222")
+@RunWith(MongoDbServerRunner.class)
+public class InsertTest extends OgmTestCase {
+
+	/**
+	 * Verify flush does not duplicate Entity on changing field, even if we are in
+	 * the same transaction that creates the Entity itself.
+	 */
+	@Test
+	public void testModifyObjectAfterPersisting() {
+		inTransaction( em -> {
+			Subject subject = new Subject( "1", "name" );
+			em.persist( subject );
+			subject.setName( "name2" );
+			em.flush();
+		} );
```

Comment:
Shouldn't you load the on object in a separate transaction and check that it's changed? How do you know that it's saved as "named2" and not just ignored?